Fix slowness on Python 3.11 when updating an existing large environment. (#12080)
pypa/pip: news/12079.bugfix.rst

> Fix slowness when using ``importlib.metadata`` and there is a large overlap between already installed and to-be-installed packages.

Reviewer: Should probably still mention 3.11, where `imporlib.metadata` is the default and impacts most people

Author: Done, and I also mention the `_PIP_USE_IMPORTLIB_METADATA` env variable that can override this.
I am happy for you to edit the news snippet directly, if that helps.

> Fix slowness when using ``importlib.metadata`` and there is a large overlap between already installed and to-be-installed packages.  This is the default in Python 3.11, though it can be overridden with the ``_PIP_USE_IMPORTLIB_METADATA`` environment variable.

Reviewer: ```suggestion
Fix slowness when using ``importlib.metadata`` (the default way for pip to read metadata in Python 3.11+) and there is a large overlap between already installed and to-be-installed packages.
```

The environment variable is intended for distro authors and let’s not mention it to end users. (And if a distro sets it, they are responsible for handling the slowdown.)
